source/helpers/selector-observer.tsx

```diff
@@ -0,0 +1,51 @@
+import mem from 'mem';
+import React from 'dom-chef';
+import {css} from 'code-tag';
+import onetime from 'onetime';
+import {ParseSelector} from 'typed-query-selector/parser';
+
+import {getSnapshotUUID} from './attach-element';
+
+const animation = 'rgh-selector-observer';
+const getListener = mem(<ExpectedElement extends HTMLElement>(seenMark: string, selector: string, callback: (element: ExpectedElement) => void) => function (event: AnimationEvent) {
+	const target = event.target as ExpectedElement;
+	if (!target.matches(selector)) {
+		return;
+	}
+
+	// Removes this specific selector’s animation once it was seen
+	target.classList.add(seenMark);
+
+	callback(target);
+});
+
+const getTag = onetime((): JSX.Element => {
+	const style = <style>{`@keyframes ${animation} {}`}</style>;
+	document.head.append(style);
+	return style;
+});
+
+export default function observe<
+	Selector extends string,
+>(
+	selectors: Selector | readonly Selector[],
+	listener: (element: ParseSelector<Selector, HTMLElement>) => void,
+	{signal}: {signal?: AbortSignal} = {},
+): void {
+	if (signal?.aborted) {
+		return;
+	}
+
+	const selector = String(selectors); // Array#toString() creates a comma-separated string
+	const seenMark = 'rgh-seen-' + getSnapshotUUID();
+	const rule = new Text(css`
+		:where(${String(selector)}):not(.${seenMark}) {
+			animation: 1ms ${animation};
+		}
+	`);
+	getTag().append(rule);
```

Comment:
Here's the `style` tag update in action, as I navigate GitHub:

https://user-images.githubusercontent.com/1402241/183035922-d7fc14a3-4d6f-471b-af90-be4d03de859c.mov